Sources/xcodeproj/Objects/BuildPhase/PBXBuildFile.swift

```diff
@@ -62,21 +62,19 @@ extension PBXBuildFile {
     /// Returns the name of the file the build file points to.
     ///
     /// - Returns: file name.
-    /// - Throws: an error if the name cannot be obtained.
-    func fileName() throws -> String? {
-        guard let fileElement: PBXFileElement = fileReference?.getObject() else { return nil }
+    func fileName() -> String? {
+        guard let fileElement: PBXFileElement = fileReference?.materialize() else { return nil }
         return fileElement.fileName()
     }
 
     /// Returns the type of the build phase the build file belongs to.
     ///
     /// - Returns: build phase type.
-    /// - Throws: an error if this method is called before the build file is added to any project.
-    func getBuildPhase() throws -> PBXBuildPhase? {
+    func getBuildPhase() -> PBXBuildPhase? {
         if let buildPhase = buildPhase {
             return buildPhase
         }
-        let projectObjects = try objects()
+        let projectObjects = objects()
```

Comment:
Developers should not call these methods on objects that are not part of any project. If that happens, the execution will abort.

Reply:
I think this is the right choice if the choices are `throw` or `abort`, since throwing really should be for runtime errors and *not* programming errors (in my interpretation of how Apple does it). However I do agree with you that an ideal situation would not allow you to call this function at all if the object is not yet there, which could be done with a careful API.